### PR TITLE
Migrate from SafetyNet to Play Integrity (VO-208)

### PR DESCRIPTION
**Pull request by acezard**
Before merging:

- [x] `cozy-flagship-app` has to be updated to the `cozy-client` version using play integration (not published yet)
- [x] integrity flow has to be tested in real world condition when the stack can verify our token

### CODE REVIEW

File: android/app/src/main/java/io/cozy/flagship/mobile/MainApplication.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import sk.kedros.playintegrity.RNGooglePlayIntegrityPackage;
```
Comment (Ldoppea) [inline android/app/src/main/java/io/cozy/flagship/mobile/MainApplication.java:7]: Why is this needed?

File: README.md [hunk 2]
```diff
@@ -133,24 +133,6 @@ See https://github.com/cozy/react-native-webview/blob/cozy_main/docs/Debugging.m
 - Then run ./scripts/build-debug-offline-apk.sh
 - The output will give you the path to the apk
 
-## How to enable Flagship certification
```
Comment (Ldoppea) [inline README.md:136]: I'm wondering if we should keep this section and talk about the google project json file instead? But I wouldn't know what to say as I don't know which part of the file is useful for the certification and which one is useful for push notifications

File: src/libs/clientHelpers/createClient.ts [hunk 3]
```diff
@@ -29,7 +28,7 @@ export const createClient = async (instance: string): Promise<CozyClient> => {
       clientKind: 'mobile',
       clientName: await getClientName(),
       shouldRequireFlagshipPermissions: true,
-      certificationConfig: { androidSafetyNetApiKey }
+      certificationConfig: {}
```
Comment (Ldoppea) [inline src/libs/clientHelpers/createClient.ts:31]: Shouldn't we put the `cloudProjectNumber` here?

Comment (acezard) [reply src/libs/clientHelpers/createClient.ts:31]: > Shouldn't we put the `cloudProjectNumber` here?

yes, has to be the prod number. for dev/local, it doesn't matter as all certification request will fail, so I'm thinking of always using the prod number regardless of env

File: README.md (hunk 2 above)
Comment (acezard) [reply README.md:136]: - [ ] Will update

File: android/app/src/main/java/io/cozy/flagship/mobile/MainApplication.java (hunk 1 above)
Comment (acezard) [reply android/app/src/main/java/io/cozy/flagship/mobile/MainApplication.java:7]: > Why is this needed?

Somehow the app wouldn't build without it, had to link manually
